### ISSUE TITLE
"macro 'parse_r' is already defined" when building LoongArch with full LTO

### ISSUE THREAD

**Issue by nathanchance** [ClangBuiltLinux/linux#1924]
When building `ARCH=loongarch` with `CONFIG_LTO_CLANG_FULL=y`, the following error is repeated several times:

```
ld.lld: error: ld-temp.o <inline asm>:105:1: macro 'parse_r' is already defined
.macro  parse_r var r
^
```

The MIPS folks hit this as well and fixed it in commit [67512a8cf5a7](https://git.kernel.org/linus/67512a8cf5a730938fdb1e48de33edfef6015e03) ("MIPS: Avoid macro redefinitions"), a copy of which LoongArch appears to need. I can work on this tomorrow, unless someone wants to beat me to it.

**Comment by nathanchance** [comment]
https://lore.kernel.org/20230905-loongarch-fix-clang-lto-parse_r-v1-1-d991ee630222@kernel.org/

**Comment by nathanchance** [comment]
Patch queued: https://git.kernel.org/chenhuacai/linux-loongson/c/8ff81bb24f68f747ab2f738c3d493b9c2cad52bf

**Comment by nathanchance** [comment]
This is actually fixed in 6.6-rc1: https://git.kernel.org/linus/8ff81bb24f68f747ab2f738c3d493b9c2cad52bf
